addons/survey/tests/test_survey.py

```diff
@@ -261,6 +261,66 @@ def test_partial_scores_simple_choice(self):
         self.assertEqual(user_input.scoring_percentage, 100)
         self.assertTrue(user_input.scoring_success)
 
+    def test_answer_time_limit_result(self):
+        test_survey = self.env['survey.survey'].create({
+            'title': 'Test This Survey',
+            'scoring_type': 'scoring_with_answers',
+            'scoring_success_min': 80.0,
+            'session_speed_rating': True,
+            'session_speed_rating_time_limit': 30,
+            'session_question_start_time': datetime.datetime(2023, 7, 7, 12, 0, 0)
+        })
+        [a_01, a_02] = self.env['survey.question.answer'].create([{
+            'value': 'In Asia',
+            'answer_score': 5.0,
+            'is_correct': True
+        }, {
+            'value': 'In Europe',
+            'answer_score': 0.0,
+            'is_correct': False
+        }])
+        q_01 = self.env['survey.question'].create({
+            'survey_id': test_survey.id,
+            'title': 'Where is india?',
+            'sequence': 1,
+            'question_type': 'simple_choice',
+            'is_time_limited': True,
+            'time_limit': 60,
+            'suggested_answer_ids': [(6, 0, (a_01 | a_02).ids)]
+        })
+        user_input = self.env['survey.user_input'].create({
+            'survey_id': test_survey.id,
+            'is_session_answer': True
+})
+        with freeze_time(lambda: datetime.datetime(2023, 7, 7, 12, 0, 30)):
+        #this answer is correct, and half of the time limit has passed
+            user_input_line1 = self.env['survey.user_input.line'].create({
+                'user_input_id': user_input.id,
+                'question_id': q_01.id,
+                'answer_type': 'suggestion',
+                'suggested_answer_id': a_01.id,
+            })
+            self.assertEqual(3.75, user_input_line1.answer_score)
```

Comment:
Better have self.assertEqual(result, expected) as it is the convention we (should) commonly use, aka
```suggestion
            self.assertEqual(user_input_line1.answer_score, 3.75)
```
(to propagate to other asserts)